crates/sui-sdk/examples/utils/example_utils.rs

```diff
@@ -0,0 +1,150 @@
+use sui_types::base_types::SuiAddress;
+use sui_types::digests::TransactionDigest;
+use sui_types::transaction::Transaction;
+
+// const SUI_FAUCET: &str = "https://faucet.devnet.sui.io/gas"; // devnet faucet
+
+const SUI_FAUCET: &str = "https://faucet.testnet.sui.io/gas"; // testnet faucet
+
+// if you use the sui-test-validator and use the local network; if it does not work, try with port 5003.
+// const SUI_FAUCET: &str = "http://127.0.0.1:9123/gas";
+
+#[allow(dead_code)]
+pub(crate) async fn sui_address_for_examples() -> Result<SuiAddress, anyhow::Error> {
+    let active_address: SuiAddress = SuiAddress::random_for_testing_only();
+    request_tokens_from_faucet(active_address).await?;
+    Ok(active_address)
+}
+
+#[allow(dead_code)]
+/// This function will try to use an active address from the wallet,
+/// and create another dummy address (in-memory) to be used in the examples.
+/// The purpose of this function is to facilitate running the examples without
+/// having to call the faucet every single time.
+/// When there is no wallet, it will create a wallet in the sui default config folder,
+/// and then create one address and it will request one coin from the faucet,
+/// for the set active address.
+pub(crate) async fn local_wallet_addresses(
+    sui: &SuiClient,
+) -> Result<Vec<SuiAddress>, anyhow::Error> {
+    let mut wallet = crate::utils::wallet::retrieve_wallet().await?;
+    // get the active address from the wallet
+    let active_address = wallet.active_address()?;
+    // get the SUI coins for this addressi
+    let coins = sui
+        .coin_read_api()
+        .get_coins(
+            active_address,
+            Some("0x2::sui::SUI".to_string()),
+            None,
+            None,
+        )
+        .await?;
+    // check if the address has coins, if not add some
+    if coins.next_cursor.is_none() {
+        request_tokens_from_faucet(active_address).await?;
+    }
+    // generate a dummy Sui address for examples only
+    let in_mem_keystore = Keystore::InMem(InMemKeystore::new_insecure_for_tests(1));
+    let addresses = in_mem_keystore.addresses();
+
+    let second_address = addresses
+        .get(0)
+        .expect("Wasn't able to generate a Sui address in-memory. Aborting.");
+
+    Ok(vec![active_address, *second_address])
+}
+
+/// Request tokens from the Faucet for the given address
+pub(crate) async fn request_tokens_from_faucet(address: SuiAddress) -> Result<(), anyhow::Error> {
+    let address_str = address.to_string();
+    let json_body = json![{
+        "FixedAmountRequest": {
+            "recipient": &address_str
+        }
+    }];
+
+    // make the request to the faucet JSON RPC API for coin
+    let client = Client::new();
+    let resp = client
+        .post(SUI_FAUCET)
+        .header("Content-Type", "application/json")
+        .json(&json_body)
+        .send()
+        .await?;
+    println!("Faucet request status: {}", resp.status());
+
+    Ok(())
+}
+
+#[allow(dead_code)]
```

Comment:
Are these annotations still required?  If you can avoid them, it will help with future maintenance.